src/Bicep.LangServer/Completions/BicepCompletionProvider.cs

```diff
@@ -168,6 +176,69 @@ private IEnumerable<CompletionItem> GetResourceTypeCompletions(SemanticModel mod
                 .ToList();
         }
 
+        private IEnumerable<CompletionItem> GetModulePathCompletions(SemanticModel model, BicepCompletionContext context)
+        {
+            if (!context.Kind.HasFlag(BicepCompletionContextKind.ModulePath)
+            || FileResolver.TryResolveModulePath(model.SyntaxTree.FileUri, ".")?.LocalPath is not string cwd
+            || new Uri(cwd) is not {} cwdUri
+            || context.EnclosingDeclaration is not ModuleDeclarationSyntax declarationSyntax)
+            {
+                return Enumerable.Empty<CompletionItem>();
+            }
+
+            // To provide intellisense before the quotes are typed
+            if (declarationSyntax.Path is not StringSyntax stringSyntax 
+            || stringSyntax.TryGetLiteralValue() is not string entered)
+            {
+                entered = "";
+            }
+
+            if (FileResolver.TryResolveModulePath(cwdUri, entered) is not {} query
+            || FileResolver.FileExists(query))
+            {
+                return Enumerable.Empty<CompletionItem>();
+            }
```

Comment:
What's the need to shortcut the main logic here? I'd expect:
1. If the user has typed the full name of the file, but has requested completions anyway, shouldn't we show the path they've already typed?
2. Although very much an edge case, it's possible to have files named `main.bicep` & `main.bicep.bicep` - in which case I'd expect completions for `main.bicep` to show both.

Reply:
+1 same question. VS code should be filtering out extra completion based on what was typed regardless.

---
In reply to: [557707767](https://github.com/Azure/bicep/pull/1236#discussion_r557707767) [](ancestors = 557707767)